client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js

```diff
@@ -806,6 +809,132 @@ const handlePrintFinished = ( orderId, siteId, dispatch, getState, hasError, lab
 	}
 };
 
+/**
+ * Generates the action that is triggered upon successful print.
+ *
+ * @param {number} count The amount of labels that were successfuly purchased and printed.
+ * @returns {Object}     TheA plain action object.
+ */
+const createPrintSuccessNotice = count => {
+	return NoticeActions.successNotice(
+		translate(
+			'Your shipping label was purchased successfully',
+			'Your %(count)d shipping labels were purchased successfully',
+			{
+				count,
+				args: { count },
+			}
+		)
+	);
+};
+
+/**
+ * Generates a action that will show an error notice upon failed label download.
+ *
+ * @param  {number}   count   The amount of labels that are being purchased.
+ * @param  {Function} onClick A callback for the retry button.
+ * @returns {Object}          The plain action that should be dispatched.
+ */
+const createPrintFailureNotice = ( count, onClick ) => {
+	const errorMessage = translate(
+		'Your shipping label was purchased successfully, but could not be printed automatically. Click "Reprint" to try again.',
+		'Your shipping labels were purchased successfully, but some of them could not be printed automatically. Click "Reprint" to try again.',
+		{ count }
+	);
+
+	return NoticeActions.errorNotice( errorMessage, {
+		id: FAILED_NOTICE_ID,
+		button: translate( 'Reprint' ),
+		onClick,
+	} );
+};
+
+/**
+ * Attempts to download and print all labels that were just generated.
+ *
+ * @param {number}   orderId  The ID of the order that is being edited.
+ * @param {number}   siteId   The ID of the current site.
+ * @param {Function} dispatch A Redux dispatcher.
+ * @param {Function} getState A function that returns the current state.
+ * @param {Array}    labels   An array of labels that should be downloaded and printed.
+ */
+function downloadAndPrint( orderId, siteId, dispatch, getState, labels ) {
+	// Backup the arguments for follow-up retries
+	const downloadArgs = arguments;
+
+	// No need to print the "Package 1 (of 1)" message if there's only 1 label
+	const labelsToPrint =
+		1 === labels.length
+			? [ { labelId: labels[ 0 ].label_id } ]
+			: labels.map( ( label, index ) => ( {
+					caption: translate( 'PACKAGE %(num)d (OF %(total)d)', {
+						args: {
+							num: index + 1,
+							total: labels.length,
+						},
+					} ),
+					labelId: label.label_id,
+			  } ) );
+
+	const { paperSize } = getShippingLabel( getState(), orderId, siteId );
+	const printUrl = getPrintURL( paperSize, labelsToPrint );
+
+	const showSuccessNotice = () => {
+		dispatch( createPrintSuccessNotice( labelsToPrint.length ) );
+	};
+
+	const retry = () => {
+		dispatch( NoticeActions.removeNotice( FAILED_NOTICE_ID ) );
+		dispatch(
+			NoticeActions.infoNotice( translate( 'Printing…' ), {
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Printing' )` *ES Score: 25*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).